internal/sdkv2provider/resource_cloudflare_list.go

```diff
@@ -94,6 +94,10 @@ func resourceCloudflareListRead(ctx context.Context, d *schema.ResourceData, met
 	d.Set("description", list.Description)
 	d.Set("kind", list.Kind)
 
+	if d.Get("ignore_inline_items").(bool) {
```

Comment:
this seems unnecessary given that during the operations we have access to the number of items present in the configuration.

Reply:
Are you suggesting that if the `cloudflare_list` resource do _not_ include any `item` attributes, we assume this is the desired behavior?  I think that would make sense for setting up a new list with externally managed list items, but it would prevent the terraform from being able to remove all the list items from a list which was populated via in-line items.  While [thats the current behavior](https://github.com/cloudflare/terraform-provider-cloudflare/issues/2294), I assume thats a bug as every terraform plan with an empty in-line item list will plan to remove all the items, which it does not, currently, do.

It would be great to not have to have another attribute on `cloudflare_list`; I feel like I'm being dense; sorry if I'm missing something obvious :-/

Reply:
to better isolate these issues, let's split the two issues here and not conflate them.

the first is focusing on when _this resource_ should make HTTP calls to save the outcome to terraform state and how that is determined. since the `cloudflare_list` resource has a list (terraform type) of `lists`, we should be able to tell whether or not we need to make the GET request for all resources to save it against this resource using `d.Get("list").([]interface{})` and running that though a `len` check. if the value is over 0, it is assumed that we want to fetch all list items for this resource.

secondly, how we manage the list items outside of this resource without blowing them all away and having conflicts. we should be able to reuse the above logic and avoid saving/fetching if the list count is less than zero since that means we're not managing the list items in the `cloudflare_list` resource.

finally, if there is a bug with how we are clearing the API when no list items are present on this resource, that is something we can triage and address. if this issue is getting the way of understanding how to proceed with the resource split, i would recommend tackling that first and then circling back to this one to avoid any overlapping work/bugs.

does that help clarify the intention here?

Reply:
If I am understanding, you are suggesting we check the number of [item](https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/internal/sdkv2provider/schema_cloudflare_list.go#L37)s defined in a `cloudflare_list`.  If it equals 0, we _assume_ we are not managing list items in-line with the list resource.  If I'm understanding that properly, that makes sense... Until the user wants to clear out a list with items managed in line.

I fear I'm missing something fundamental because with the above logic, you could manage list items inline as long as you didn't want to use terraform to delete all list items.  If you use the `cloudflare_list` to deploy items in-line, you'd be unable to use the `cloudflare_list` resource to delete all items as clearing out the in line item definitions would cause the resource to ignore any deployed list items.  It might not be necessary functionality, I think it would be un-intuitive (to clear out the in-line items and have them NOT get deleted)

...unless there is way to inspect state and know that there _was_ `items` defined that are no longer there

Sorry if I'm still not understanding or not being clear in what i'm thinking

Reply:
> If it equals 0, we _assume_ we are not managing list items in-line with the list resource

correct and this assumption is *only* used for whether or not to persist to the state file. 

> [...] Until the user wants to clear out a list with items managed in line.
> ...unless there is way to inspect state and know that there was items defined that are no longer there

for the case where we would want to clear out the existing inline managed rules, you'll need to do a check using `d.GetChange` (similar to https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/internal/sdkv2provider/resource_cloudflare_teams_list.go#L120-L134). this approach will allow you to determine if list items were previously managed and whether you should initate removing the inline managed items. at that point, if a user wanted to manage them separately, they will need to already have the resources ready to apply once the list is empty.

Reply:
This is the piece i was missing; thank you for the direction.  

Update inbound shortly!